lib/ProductOpener/Display.pm

```diff
@@ -7025,7 +7025,7 @@ sub display_page($) {
 		$canon_description = lang("site_description");
 	}
 	my $canon_image_url = "";
-	my $canon_url = $formatted_subdomain;
+	my $canon_url = 'http://world.productopener.localhost';
```

Comment:
There are 2 issues:

1. http://world.productopener.localhost is a url that works only on your developer machine. In practice, we want to have the domain https://world.openfoodfacts.org/ in production, https://world.openfoodfacts.dev on the dev server etc.

2. we only want to change the subdomain from uk/ca/us.openfoodfacts.org etc. to world.openfoodfacts.org for the canon_url of pages that are both product pages, and that are in English.

In Display.pm, in the display_product() function, there is this line:

$request_ref->{canon_url} = product_url($product_ref);

That's where you want to change the subdomain (if the language is in English :  $lc == 'en').

Note that Display.pm sets a $formatted_subdomain (e.g. uk.openfoodfacts.org or uk.productopener.localhost on docker) and a $world_subdomain (world.openfoodfacts.org or world.productopener.localhost)